### PR TITLE
Added demo for character count component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
#160

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-demos](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-demos)

File: src/app/docs/character-count/character-count-docs.component.html [hunk 1]
```diff
@@ -7,6 +7,52 @@
     The character count indicator component extends a text input to apply a character limit and display an indicator with the number of characters entered and the character limit.
   </sky-docs-demo-page-summary>
 
+  <sky-docs-demo>
+    <div
+      style="width: 50%;"
```
Comment (Blackbaud-TrevorBurch) [inline src/app/docs/character-count/character-count-docs.component.html:12]: What is our "standard" here? I see differences accross our components:
- Datepicker is set to `400px` 
- Country Field goes 100%
- Here we are doing 50%

Should we have a standard for this?

File: src/app/docs/character-count/character-count-docs.component.html [hunk 2]
```diff
@@ -192,4 +238,11 @@
 
   </sky-docs-design-guidelines>
 
+  <sky-docs-code-examples>
+    <sky-docs-code-example
+      sourceCodePath="src/app/public/plugin-resources/code-examples/character-count"
```
Comment (Blackbaud-TrevorBurch) [inline src/app/docs/character-count/character-count-docs.component.html:243]: I think we may need to discuss if we should have a standard pattern here as I've seen various versions of calling this nothing (as you have it), calling it `Basic setup`, and calling this `Character count`. Do you know if we have a standard we should be following?

File: src/app/public/plugin-resources/code-examples/character-count/character-count-demo.component.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<div
+  style="width: 50%;"
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/plugin-resources/code-examples/character-count/character-count-demo.component.html:2]: Do we need this div here for the code example? I know it is just the demo code but there isn't truly a reason for it to be here for the example unless I'm missing something.

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
@Blackbaud-AlexKingman just a couple very minor questions.

File: src/app/docs/character-count/character-count-docs.component.html (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/docs/character-count/character-count-docs.component.html:243]: I've been following this pattern - if it has multiple demos, put them in folders. If it has just one demo, then there's no need for folders. How does that sound?

File: src/app/public/plugin-resources/code-examples/character-count/character-count-demo.component.html (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/character-count/character-count-demo.component.html:2]: I just based this on the design spec (see issue for link).

Comment (Blackbaud-TrevorBurch) [reply src/app/public/plugin-resources/code-examples/character-count/character-count-demo.component.html:2]: Right. I get that for the demo - just didn't know if it needed to be there for the code example. Not a big deal either way.

File: src/app/docs/character-count/character-count-docs.component.html (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/docs/character-count/character-count-docs.component.html:243]: Lets discuss in flex time but I'm not too worried about it.

File: src/app/docs/character-count/character-count-docs.component.html (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/docs/character-count/character-count-docs.component.html:12]: That's fine for this. We can adjust if needed later.

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**